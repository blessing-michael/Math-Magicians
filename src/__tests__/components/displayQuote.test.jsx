import React from 'react';
import { render } from '@testing-library/react';
import DisplayQuote from '../../components/displayQuote';

test('renders loader when isLoading is true', () => {
  const { container } = render(<DisplayQuote isLoading />);
  expect(container).toMatchSnapshot();
});

test('renders error message when error is not null', () => {
  const error = 'Failed To Load Quote';
  const { container } = render(<DisplayQuote error={error} />);
  expect(container).toMatchSnapshot();
});

test('renders quote and author when quote is not null', () => {
  const quote = [{ quote: 'Test Quote', author: 'Test Author' }];
  const { container } = render(<DisplayQuote quote={quote} />);
  expect(container).toMatchSnapshot();
});

test('renders "No Quote Found" when quote is null', () => {
  const { container } = render(<DisplayQuote quote={null} />);
  expect(container).toMatchSnapshot();
});
