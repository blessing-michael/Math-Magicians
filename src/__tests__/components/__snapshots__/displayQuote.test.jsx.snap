// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "No Quote Found" when quote is null 1`] = `
<div>
  <div
    class="loader"
  >
    <img
      alt="loader"
      class="loader-img"
      src="loader.svg"
    />
    <p
      class="loader-tag"
    >
      We are getting you a quote
      <br />
      Please wait...
    </p>
  </div>
</div>
`;

exports[`renders error message when error is not null 1`] = `
<div>
  <div
    class="loader"
  >
    <img
      alt="loader"
      class="loader-img"
      src="loader.svg"
    />
    <p
      class="loader-tag"
    >
      We are getting you a quote
      <br />
      Please wait...
    </p>
  </div>
</div>
`;

exports[`renders loader when isLoading is true 1`] = `
<div>
  <div
    class="loader"
  >
    <img
      alt="loader"
      class="loader-img"
      src="loader.svg"
    />
    <p
      class="loader-tag"
    >
      We are getting you a quote
      <br />
      Please wait...
    </p>
  </div>
</div>
`;

exports[`renders quote and author when quote is not null 1`] = `
<div>
  <div
    class="loader"
  >
    <img
      alt="loader"
      class="loader-img"
      src="loader.svg"
    />
    <p
      class="loader-tag"
    >
      We are getting you a quote
      <br />
      Please wait...
    </p>
  </div>
</div>
`;
